src/Microsoft.TemplateEngine.Edge/Installers/NuGet/NuGetManagedTemplatePackage.cs

```diff
@@ -235,25 +170,16 @@ public bool IsLocalPackage
 
         public IReadOnlyDictionary<string, string> GetDetails()
         {
-            Dictionary<string, string> details = new Dictionary<string, string>();
-            if (!string.IsNullOrWhiteSpace(Author))
-            {
-                details[AuthorKey] = Author!;
-            }
-            if (!string.IsNullOrWhiteSpace(Owners))
-            {
-                details[OwnersKey] = Owners!;
-            }
-            if (!string.IsNullOrWhiteSpace(Trusted))
-            {
-                details[TrustedKey] = Trusted!;
-            }
-            if (!string.IsNullOrWhiteSpace(NuGetSource))
-            {
-                details[NuGetSourceKey] = NuGetSource!;
-            }
+            var details = new Dictionary<string, string>();
+
+            details.TryAdd(AuthorKey, Author ?? string.Empty, InsertionCondition);
```

Comment:
should insertion condition be `!string.IsNullOrWhitespace(entry)` (applies to all lines below). 
also it doesn't seem to do exactly what was done before.
`InsertionCondition` is not very representative name.  I'm not sure if the suggested code doesn't overcomplicate it.

```suggestion
            details.TryAdd(AuthorKey, Author, !string.IsNullOrWhiteSpace(Author));
```

Reply:
The `TryAdd` method has a Predicate on the last parameter, that's why we got the `InsertCondition`. I will rename it to `ValidStringCheck`, but I think it is an interesting function to create a pattern for classes that use the Dictionary.